src/coord/src/coord.rs

```diff
@@ -588,23 +588,22 @@ impl Coordinator {
         }
     }
 
-    fn message_worker(
-        &mut self,
-        dataflow::Response {
-            worker_id: _,
-            message,
-        }: dataflow::Response,
-    ) {
+    fn message_worker(&mut self, dataflow::Response { worker_id, message }: dataflow::Response) {
         match message {
             WorkerFeedback::PeekResponse(conn_id, response) => {
                 // We use an `if let` here because the peek could have been cancelled already.
```

Comment:
This comment isn't true. A cancel will not eagerly drop this. Maybe it should?

Reply:
ty for catching; this is essentially the same as the `else` condition for having already received a response from a worker on the pending peek, so I'm logging this as a warning. As mentioned above, Frank's proposed design is to await messages from dataflow in all cases, which I think this is sane and more likely to point us to buggy code.